src/posts/lucky_1.0.0-rc1.cr

```diff
@@ -0,0 +1,239 @@
+    uncertianty, the 1.0.0-rc1 release opens up the possibility to use [Vite](https://vitejs.dev/)
+    while keeping with the type-safe traditions Lucky brings.
+
+    So why even include a javascript build system with Lucky?
+
+    One of Lucky's goals is to be type-safe and help catch bugs in development. This
+    is done many ways, but one of the most important ways is when it comes to loading
+    your assets in production. By utilizing a generated manifest file, Crystal can
+    generate an internal Hash at compile-time that can be referenced in the app.
+
+    This means in development, you can do `img src: asset("images/logo.svg")`, and
+    Lucky knows this file is located at `public/assets/images/logo.svg`. In production,
+    you may need to use a cache buster, or even a CDN where the filename may change to
+    say `assets.mysite.com/images/logo-abc123xyz.svg`. If this file is missing, then
+    your app will not compile. That means that last second production deploy you're
+    doing on a Friday before a 3-day weekend won't have you back in the "office"
+    doing another deploy because the file isn't showing.
+
+    ### Email layouts and callbacks
+
+    Carbon 0.3.0 has been released with a few neat updates. The first being
+    the ability to set email layouts. Just like your standard views, you may
+    have a very similar wrapper on your emails like contact footer info, or
+    your logo.
+
+    To specify your layout, you'll add the `layout` macro, and pass the name
+    of the folder where your layout template file is located.
+
+    ```crystal
+    # src/emails/welcome_email.cr
+    class WelcomeEmail < BaseEmail
+      def initialize(@recipient : Carbon::Emailable)
+      end
+
+      to @recipient
+      from "info@myapp.com"
+      subject "Welcome to the site!"
+      layout my_email_layout
+      templates html, text
+    end
+    ```
+
+    Then in your templates, you can add `my_email_layout/layout.ecr`,
+    and use the `content` method to yield your email body
+
+    ```
+    # src/emails/templates/my_email_layout/layout.ecr
+    <h1>Our Email</h1>
+
+    <%= content %>
+
+    <div>footer</div>
+
+    # src/emails/templates/welcome_email/html.ecr
+    <p>Welcome to the site!</p>
+    ```
+
+    Emailing can be tricky. Sometimes it's hard to track emails,
+    and other times you're dealing with unsubscribes. To help with
+    these, we've added `before_send` and `after_send` callbacks,
+    as well as a `deliverable : Bool` property. When `deliverable`
+    is set to `false`, the email won't be sent.
+
+    ```crystal
+    class FriendIsLiveEmail < BaseEmail
+      def initialize(@recipient : Carbon::Emailable, @friend : User)
+      end
+
+      before_send do
+        if !@recipient.wants_email_from_friend?(@friend)
+          self.deliverable = false
+        end
+      end
+
+      # the response is from the email service (i.e. sendgrid, etc...)
```

Comment:
Are there docs we could link to for this? I think this may be hard to understand without some further docs.

Reply:
Not really.. it's going to be adapter specific since the response is `settings.adapter.deliver_now(self)`. So like for sendgrid it'll be the client response from the API, but for SMTP would be a tuple with the status code and message...